### ISSUE TITLE
WindowIdentifier::from_native() may only be called once for Wayland GTK windows

### ISSUE THREAD

**Issue by Cogitri** [bilelmoussaoui/ashpd#20]
Hello,

it seems like WindowIndentifier::from_native() can only be called once, otherwise the following GCritical is emitted:

```
(dev.Cogitri.Health:21): Gdk-CRITICAL **: 12:22:08.835: gdk_wayland_toplevel_export_handle: assertion '!impl->display_server.xdg_exported' failed
```

This GCritical origins from `gdk_wayland_toplevel_export_handle`. The documentation for that function says:

```
* gdk_wayland_toplevel_export_handle:
[...]
* It is an error to call this function on a surface that is already
* exported.
```

Not sure what would be the best way to handle this, should we just put a notice in the docs?

**Comment by bilelmoussaoui** [comment]
Yes that's expected because the handle is unexported when the WindowIdentifier is dropped. 

I wonder what would be the best way forward here, but addikg a comment at least seems fine to me

**Comment by Cogitri** [comment]
Alright, I’ll submit a PR later today.

Another question: To work around this I currently just hold onto the WindowIndentifier in a OnceCell that’s initialised when the window is created. Is that fine or should I always re-request the WindowIdentifier?

**Comment by bilelmoussaoui** [comment]
I would say you can keep only once instance around by wrapping it in a OnceCell. I don't know if we can do something about it on the crate side?

**Comment by Cogitri** [comment]
Hm, I don’t really think we can other than hacky ways like having a global HashMap<WeakRef<Native>, WindowIdentifier> to remember the WindowIdentifier

**Comment by bilelmoussaoui** [comment]
Ah yeah indeed. That would be very painful to do. I guess going with a comment would be good enough for now.

Feel free to open another issue regarding the usage of the Default implementation of WindowIdentifier in some of the portals

**Comment by A6GibKm** [comment]
Cant we do a static Once that holds the identifier?
